worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/IngestionBatchWorkerPlugin.java

```diff
@@ -0,0 +1,274 @@
+                throw new RuntimeException("Invalid Batch Id Exception! " + ex.getMessage());
+            }
+        } else if (batchDefinition.contains("|")) {
+            log.debug("Multiple Batches received: " + batchDefinition);
+            handleMultipleBatchIds(batchDefinition, batchWorkerServices);
+        } else {
+            log.debug("Single Batch received: " + batchDefinition);
+            handleSingleBatchId(batchDefinition, batchWorkerServices);
+        }
+
+    }
+
+    private void handleMultipleBatchIds(final String batchIds, final BatchWorkerServices batchWorkerServices)
+    {
+        final String tenantId = extractTenantId(batchIds);
+        final String[] batchesSplit = extractBatchIds(batchIds);
+        for (final String batch : batchesSplit) {
+            batchWorkerServices.registerBatchSubtask(tenantId + "/" + batch);
+        }
+    }
+
+    private void handleSingleBatchId(final String batchId, final BatchWorkerServices batchWorkerServices)
+    {
+        try {
+            final TenantId tenantId = new TenantId(extractTenantId(batchId));
+            final BatchId batchIdExtracted = new BatchId(extractBatchIds(batchId)[0]);
+            final Path pathOfSubBatches = fileSystemProvider.getPathForBatch(tenantId, batchIdExtracted);
+            final String[] extensions = {"batch"};
+            final Collection<File> subbatchesFiles = FileUtils.listFiles(pathOfSubBatches.toFile(), extensions, false);
+            
+            for (final File subbatch : subbatchesFiles) {
+                log.debug("Batch file found: " + FilenameUtils.getName(subbatch.getAbsolutePath()));
+                batchWorkerServices.registerBatchSubtask("subbatch:" + tenantId.getValue() + "/" + batchIdExtracted.getValue()
+                    + "/" + FilenameUtils.getName(subbatch.getAbsolutePath()));
+            }
+        } catch (InvalidBatchIdException | InvalidTenantIdException ex) {
+            log.error("Exception while handling single batch id: " + ex.getMessage());
+            throw new RuntimeException("Exception while handling a single batch id: " + ex.getMessage());
+        }
+
+    }
+
+    private void handleSubbatch(final String subbatch, final BatchWorkerServices batchWorkerServices,
+                                final Map<String, String> taskMessageParams) throws InvalidBatchIdException
+    {
+        final Subbatch subbatchObj = extractSubbatch(subbatch);
+        final Path pathOfSubBatches = fileSystemProvider.getPathForBatch(subbatchObj.getTenantId(), subbatchObj.getBatchId());
+        final Path batchFileName = Paths.get(pathOfSubBatches.toString(), subbatchObj.getFileName());
+        log.debug("I am going to read each line of: " + batchFileName);
+        try (Stream<String> lines = Files.lines(batchFileName)) {
+            lines.forEach(line -> {
+                try {
+                    final DocumentWorkerDocumentTask document = mapper.readValue(line, DocumentWorkerDocumentTask.class);
```

Comment:
Split this out into a separate method:

    private DocumentWorkerDocumentTask createDocument(final String line)